### PR TITLE
tidb-functions: add info about special start_key/end_key values

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

```sql
sql> SELECT TABLE_NAME, REGION_ID, START_KEY, END_KEY FROM information_schema.TIKV_REGION_STATUS WHERE TABLE_NAME='stock' AND IS_INDEX=0;
+------------+-----------+--------------------------------------------------------+--------------------------------------------------------+
| TABLE_NAME | REGION_ID | START_KEY                                              | END_KEY                                                |
+------------+-----------+--------------------------------------------------------+--------------------------------------------------------+
| stock      |        74 | 7480000000000000FF4700000000000000F8                   | 7480000000000000FF475F728000000000FF029DF00000000000FA |
| stock      |        72 | 7480000000000000FF475F728000000000FF0D7ECA0000000000FA | 7480000000000000FF4900000000000000F8                   |
| stock      |        78 | 7480000000000000FF475F728000000000FF063E890000000000FA | 7480000000000000FF475F728000000000FF09E0350000000000FA |
| stock      |        76 | 7480000000000000FF475F728000000000FF029DF00000000000FA | 7480000000000000FF475F728000000000FF063E890000000000FA |
| stock      |        80 | 7480000000000000FF475F728000000000FF09E0350000000000FA | 7480000000000000FF475F728000000000FF0D7ECA0000000000FA |
+------------+-----------+--------------------------------------------------------+--------------------------------------------------------+
5 rows in set (0.0144 sec)

sql> SELECT TIDB_DECODE_KEY('7480000000000000FF4700000000000000F8');
+---------------------------------------------------------+
| TIDB_DECODE_KEY('7480000000000000FF4700000000000000F8') |
+---------------------------------------------------------+
| 7480000000000000FF4700000000000000F8                    |
+---------------------------------------------------------+
1 row in set, 1 warning (0.0013 sec)
Warning (code 1105): invalid record/index key: 748000000000000047

sql> SELECT TIDB_DECODE_KEY('7480000000000000FF475F728000000000FF0D7ECA0000000000FA');
+---------------------------------------------------------------------------+
| TIDB_DECODE_KEY('7480000000000000FF475F728000000000FF0D7ECA0000000000FA') |
+---------------------------------------------------------------------------+
| {"_tidb_rowid":884426,"table_id":"71"}                                    |
+---------------------------------------------------------------------------+
1 row in set (0.0013 sec)
```

```
sql> SELECT TABLE_NAME, REGION_ID, TIDB_DECODE_KEY(START_KEY) sk, TIDB_DECODE_KEY(END_KEY) ek FROM information_schema.TIKV_REGION_STATUS WHERE TABLE_NAME='stock' AND IS_INDEX=0 ORDER BY IF(JSON_VALID(sk),sk->'$."_tidb_rowid"',0);
+------------+-----------+----------------------------------------+----------------------------------------+
| TABLE_NAME | REGION_ID | sk                                     | ek                                     |
+------------+-----------+----------------------------------------+----------------------------------------+
| stock      |        74 | 7480000000000000FF4700000000000000F8   | {"_tidb_rowid":171504,"table_id":"71"} |
| stock      |        76 | {"_tidb_rowid":171504,"table_id":"71"} | {"_tidb_rowid":409225,"table_id":"71"} |
| stock      |        78 | {"_tidb_rowid":409225,"table_id":"71"} | {"_tidb_rowid":647221,"table_id":"71"} |
| stock      |        80 | {"_tidb_rowid":647221,"table_id":"71"} | {"_tidb_rowid":884426,"table_id":"71"} |
| stock      |        72 | {"_tidb_rowid":884426,"table_id":"71"} | 7480000000000000FF4900000000000000F8   |
+------------+-----------+----------------------------------------+----------------------------------------+
5 rows in set, 2 warnings (0.0133 sec)
Warning (code 1105): invalid record/index key: 748000000000000047
Warning (code 1105): invalid record/index key: 748000000000000049

```

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### CODE REVIEW

**Comment by kolbe** [comment]
@dveeden this is great, but maybe you could also submit a patch to tidb to make tidb_decode_key do something more sensible with these values 🙂

**Comment by dveeden** [comment]
@kolbe yes, that's my plan.

**Review by kolbe — APPROVED** [review]
LGTM

**Comment by dveeden** [comment]
cc @morgo

**Comment by qiancai** [comment]
Removed the **needs-cherry-pick-release-6.3** label because the v6.3 docs have been archived at https://docs-archive.pingcap.com/zh/tidb/v6.3 and will no longer receive new updates.

**Comment by qiancai** [comment]
Removed the needs-cherry-pick-release-6.4 label because the v6.4 docs have been archived at https://docs-archive.pingcap.com/tidb/v6.4 and will no longer receive new updates.

**Comment by qiancai** [comment]
Removed the needs-cherry-pick-release-6.6 label because the v6.6 docs have been archived at https://docs-archive.pingcap.com/tidb/v6.6 and will no longer receive new updates.

File: functions-and-operators/tidb-functions.md [hunk 1]
```diff
@@ -102,6 +102,29 @@ select tidb_decode_key('7480000000000000FF3E5F720400000000FF0000000601633430FF33
 1 row in set (0.001 sec)
 ```
 
+The first region of a table will start with a key that only has the `table_id` of the table. The last region of the table ends with a `table_id` of the next table. Any regions in between will have longer keys that includes a `_tidb_rowid` or `handle`.
```
Comment (mjonss) [inline functions-and-operators/tidb-functions.md:105]:
```suggestion
The first region of a table will start with a key that only has the `table_id` of the table. The last region of the table ends with `table_id` + 1. Any regions in between will have longer keys that includes a `_tidb_rowid` or `handle`.
```

File: functions-and-operators/tidb-functions.md [hunk 2]
```diff
@@ -102,6 +102,29 @@ select tidb_decode_key('7480000000000000FF3E5F720400000000FF0000000601633430FF33
...
+```sql
+SELECT TABLE_NAME, START_KEY, END_KEY FROM information_schema.TIKV_REGION_STATUS WHERE TABLE_NAME='stock' AND IS_INDEX=0;
+```
+
+```sql
++------------+--------------------------------------------------------+--------------------------------------------------------+
+| TABLE_NAME | START_KEY                                              | END_KEY                                                |
++------------+--------------------------------------------------------+--------------------------------------------------------+
+| stock      | 7480000000000000FF4700000000000000F8                   | 7480000000000000FF475F728000000000FF029DF00000000000FA |
+| stock      | 7480000000000000FF475F728000000000FF0D7ECA0000000000FA | 7480000000000000FF4900000000000000F8                   |
```
Comment (mjonss) [inline functions-and-operators/tidb-functions.md:118]: Should the END_KEY be 7480000000000000FF4800000000000000F8 (i.e. table_id + 1)? Internally I would assume we are skipping an id, effectively doing table_id + 2?

Comment (dveeden) [reply functions-and-operators/tidb-functions.md:145]:
```
sql> SELECT DB_NAME, TABLE_ID MOD 2, COUNT(*) FROM information_schema.TIKV_REGION_STATUS WHERE TABLE_ID GROUP BY DB_NAME, TABLE_ID MOD 2 ORDER BY 1,2;
+--------------------+----------------+----------+
| DB_NAME            | TABLE_ID MOD 2 | COUNT(*) |
+--------------------+----------------+----------+
| INFORMATION_SCHEMA |              0 |       44 |
| INFORMATION_SCHEMA |              1 |       45 |
| METRICS_SCHEMA     |              0 |      320 |
| METRICS_SCHEMA     |              1 |      320 |
| PERFORMANCE_SCHEMA |              0 |       16 |
| PERFORMANCE_SCHEMA |              1 |       18 |
| mysql              |              0 |      112 |
| mysql              |              1 |        6 |
| test               |              0 |       19 |
+--------------------+----------------+----------+
9 rows in set (0.0071 sec)
```
Here the `test` schema only has even table_id's.

**Review by mjonss — APPROVED** [review]

File: functions-and-operators/tidb-functions.md (hunk 2 above)
Comment (mjonss) [reply functions-and-operators/tidb-functions.md:145]: Reported as https://github.com/pingcap/tidb/issues/48725.

File: functions-and-operators/tidb-functions.md [hunk 3]
```diff
@@ -129,6 +129,29 @@ select tidb_decode_key('7480000000000000FF3E5F720400000000FF0000000601633430FF33
 1 row in set (0.001 sec)
 ```
 
+The first region of a table will start with a key that only has the `table_id` of the table. The last region of the table ends with `table_id` + 1. Any regions in between will have longer keys that includes a `_tidb_rowid` or `handle`.
+
+{{< copyable "sql" >}}
+
+```sql
+SELECT TABLE_NAME, START_KEY, END_KEY FROM information_schema.TIKV_REGION_STATUS WHERE TABLE_NAME='stock' AND IS_INDEX=0;
```
[the remark this answers is not in the document]
Comment (dveeden) [reply functions-and-operators/tidb-functions.md:127]: Not sure if that's good for this section. I think it is, but I need to double check before we can change this.

**Review by Oreoxmt — APPROVED** [review]

**Review by ran-huang — APPROVED** [review]

File: functions-and-operators/tidb-functions.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (dveeden) [reply functions-and-operators/tidb-functions.md]: Changed this in 975edbcf520a7041450d7088c14c099e81acdd02

[the remark this answers is not in the document]
Comment (dveeden) [reply functions-and-operators/tidb-functions.md]: PTAL @Oreoxmt

**Pull request merged**